### PR TITLE
added tracie brach with data and example evaluation runs

### PR DESCRIPTION
**Pull request by yakazimir**

### CODE REVIEW

File: tracie/data_public/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+This directory contains the files `train.jsonl`, `dev.jsonl` and `test.jsonl`. Please note that `train.jsonl` and `dev.jsonl` are the same file, since we do not have an explicit `dev` split (for our experiments, we use a portion of the training set to tune models on `train.jsonl`).
+
+Each example in the file looks like the following:
+```
+{
+    "query": "event: Tom's teeth are crooked ends before he has braces on for a while",
+    "story": "Tom needed to get braces. He was afraid of them. The dentist assured him everything would be fine. Tom had them on for a while. Once removed he felt it was worth it.",
+    "gold_label": "contradiction"}
+```
+and consists of three fields: `story` (or the premise); `query` (or hypothesis) and `gold_label` (the inference label). The testing labels are hidden and replaced with `"-"`
```
Comment (aimichal) [inline tracie/data_public/README.md]:
```suggestion
and consists of three fields: `story` (the premise); `query` (the hypothesis) and `gold_label` (the inference label). The testing labels are hidden and replaced with `"-"`
```

File: tracie/data_public/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+This directory contains the files `train.jsonl`, `dev.jsonl` and `test.jsonl`. Please note that `train.jsonl` and `dev.jsonl` are the same file, since we do not have an explicit `dev` split (for our experiments, we use a portion of the training set to tune models on `train.jsonl`).
```
Comment (aimichal) [inline tracie/data_public/README.md]: If train.jsonl and dev.jsonl are the same, and dev.jsonl isn't used anywhere in this code, can we just keep train.jsonl (which is used)?

File: tracie/evaluator/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+## Example
+```
+%python3 evaluator/evaluator.py --question_answers data_public/train.jsonl --predictions data_public/train_predictions.txt --output metrics.json
+
+% cat metrics.json
+{"total_acc": 0.5012809564474808, "start_acc": 0.500945179584121, "end_acc": 0.5015576323987538, "story_em": 0.0}
+```
+
+This uses a dummy train prediction file called `train_predictions.txt` from the training file `train.jsonl` (which predicts entailments for each example), which consists of one label prediction for line:
```
Comment (aimichal) [inline tracie/evaluator/README.md]:
```suggestion
This uses a dummy train prediction file called `train_predictions.txt` from the training file `train.jsonl` (which predicts entailments for each example), which consists of one label prediction per line (that is, per example):
```

File: tracie/evaluator/evaluator.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                correct_end += 1
+    s_total = 0
+    s_correct = 0
+    for key in story_prediction_map:
+        s_total += 1
+        cv = True
+        for v in story_prediction_map[key]:
+            cv = cv and v
+        if cv:
+            s_correct += 1
```
Comment (aimichal) [inline tracie/evaluator/evaluator.py]:
```suggestion
        if all(story_prediction_map[key]):
            s_correct += 1
```

File: tracie/evaluator/evaluator.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+                "start_acc": start_acc,
+                "end_acc"  : end_acc,
+                "story_em" : story_em,
+            }
+        ))
+
+if __name__ == "__main__":
+    main()
+
+#print(evaluate("iid/test.jsonl", "sample_prediction_iid.txt"))
```
Comment (aimichal) [inline tracie/evaluator/evaluator.py]:
```suggestion
```

**Review by aimichal — COMMENTED** [review]
Looks pretty good. Have a look at the comments, it's nothing major, just some thoughts.

File: tracie/data_public/README.md (hunk 2 above)
Comment (yakazimir) [reply tracie/data_public/README.md]: yeh, that is fine.

Comment (yakazimir) [reply tracie/data_public/README.md]: can you go ahead and remove the dev file then?

**Comment by aimichal** [comment]
@yakazimir Alrighty, I made some changes. Please, have a look at the READMEs and files created. Browse at: https://github.com/allenai/aristo-leaderboard/tree/tracie/tracie

Specifically, I renamed data_public to data (to simplify) and created a new file data/test-dummy-predictions.txt that can be submitted to the Leaderboard (when it's available.)

**Comment by yakazimir** [comment]
looks great to me, thanks also for adding the test prediction example and bette explaining the overall protocol.

**Pull request merged**